### ISSUE TITLE
Cannot reproduce results on resnet34.

### ISSUE THREAD

**Issue by ia3leonidshad** [MingSun-Tse/Regularization-Pruning#4]
Hi,

I am trying to reproduce ImageNet Resnet34 Greg-2 pruning results.
Baseline acc is 73.312%.
GPU: RTX 3070

CUDA_VISIBLE_DEVICES=0 python main.py --dataset imagenet --arch resnet34 --pretrained --method GReg-2 --screen --stage_pr [0,0.5,0.6,0.4,0,0] --skip_layers [1.0,2.0,2.3,3.0,3.5] --project GReg-2__resnet34__imagenet__1.32x_pr0.50.60.4 --data_path /home/leonid/Documents/

Intermediate output:

_[120331 4071 2022/02/21-13:51:25] Iter = 48200 [prune_state = update_reg, method = GReg-2] ----------------------------------------
[120331 4071 2022/02/21-13:51:25] [3] Update reg for layer 'module.layer1.1.conv1'. Pr = 0.5. Iter = 48200
[120331 4071 2022/02/21-13:51:25]     prune stage, push the pruned (reg = 0.87404) to zero; for kept weights, reg = -0.00010
[120331 4071 2022/02/21-13:51:25]     mag_ratio 916.3661 mag_ratio_momentum 925.2601
[120331 4071 2022/02/21-13:51:25]     for kept weights, original_kept_w_mag 0.028890, now_kept_w_mag 0.025137 ratio_now_over_original 0.8701
[120331 4071 2022/02/21-13:51:25]     reg_status: min = -0.00010 ave = 0.43697 max = 0.87404
[120331 4071 2022/02/21-13:51:25] After optim update, ave_abs_weight: 0.0107684601 current_train_loss: 1.0688 current_train_acc: 0.7344
[120331 4071 2022/02/21-13:51:25] speed = 11.9691 iter/s
[120331 4071 2022/02/21-13:51:30] ==> [3] Just finished 'update_reg'. Iter = 48255
[120331 4071 2022/02/21-13:51:30] ==> All layers just finished 'update_reg', go to 'stabilize_reg'. Iter = 48255
[120331 4071 2022/02/21-13:51:34]
[120331 4071 2022/02/21-13:51:34] Iter = 48300 [prune_state = stabilize_reg, method = GReg-2] ----------------------------------------
[120331 4071 2022/02/21-13:51:34] After optim update, ave_abs_weight: 0.0107686277 current_train_loss: 1.0065 current_train_acc: 0.7969
[120331 4071 2022/02/21-13:51:34] speed = 10.9782 iter/s_

At the end of the pruning accuracy is 69.8040 %, but after first epoch of fine-tuning it dropped to 65.1520% and never recovered.

_[120331 4071 2022/02/21-15:14:47] Iter = 88200 [prune_state = stabilize_reg, method = GReg-2] ----------------------------------------
[120331 4071 2022/02/21-15:14:47] After optim update, ave_abs_weight: 0.0108080273 current_train_loss: 1.6889 current_train_acc: 0.6094
[120331 4071 2022/02/21-15:14:47] speed = 11.3965 iter/s
[120331 4071 2022/02/21-15:14:52] [64, 64, 64, 32, 64, 32, 64, 128, 128, 51, 128, 51, 128, 128, 128, 256, 256, 153, 256, 153, 256, 153, 256, 153, 256, 256, 256, 512, 512, 512, 512, 512, 512]
[120331 4071 2022/02/21-15:14:52] 'stabilize_reg' is done. Pruned, go to 'finetune'. Iter = 88255
[120331 4071 2022/02/21-15:14:52] ==> n_params_original_v2: 21.7806M, n_flops_original_v2:  7.3275G
[120331 4071 2022/02/21-15:14:52] ==> n_params_now_v2:      19.4536M, n_flops_now_v2:       5.5645G
[120331 4071 2022/02/21-15:14:52] ==> reduction ratio -- params: 10.68% (compression 1.12x), flops: 24.06% (speedup 1.32x)
[120331 4071 2022/02/21-15:16:00]    Acc1 69.8040 Acc5 89.4300 Loss_test 1.2277 | Acc1_train -1.0000 Acc5_train -1.0000 Loss_train -1.0000 | (test_time 67.91s) Just got pruned model, about to finetune
[120331 4071 2022/02/21-15:16:00] ==> Start to finetune: using SGD optimizer
[120331 4071 2022/02/21-15:16:00] ==> Set lr = 0.01 @ Epoch 0
[ 0]              module.conv1 -- mag_old = 0.0710, mag_now = 0.0740 (1.04)
[ 1]     module.layer1.0.conv1 -- mag_old = 0.0243, mag_now = 0.0262 (1.08)
[ 2]     module.layer1.0.conv2 -- mag_old = 0.0241, mag_now = 0.0259 (1.07)
[ 3]     module.layer1.1.conv1 -- mag_old = 0.0256, mag_now = 0.0298 (1.17)
[ 4]     module.layer1.1.conv2 -- mag_old = 0.0241, mag_now = 0.0291 (1.21)
[ 5]     module.layer1.2.conv1 -- mag_old = 0.0258, mag_now = 0.0289 (1.12)
[ 6]     module.layer1.2.conv2 -- mag_old = 0.0231, mag_now = 0.0274 (1.19)
[ 7]     module.layer2.0.conv1 -- mag_old = 0.0245, mag_now = 0.0260 (1.06)
[ 8]     module.layer2.0.conv2 -- mag_old = 0.0185, mag_now = 0.0197 (1.07)
[ 8] module.layer2.0.downsample.0 -- mag_old = 0.0413, mag_now = 0.0430 (1.04)
[ 9]     module.layer2.1.conv1 -- mag_old = 0.0168, mag_now = 0.0193 (1.15)
[10]     module.layer2.1.conv2 -- mag_old = 0.0167, mag_now = 0.0199 (1.19)
[11]     module.layer2.2.conv1 -- mag_old = 0.0182, mag_now = 0.0202 (1.11)
[12]     module.layer2.2.conv2 -- mag_old = 0.0167, mag_now = 0.0194 (1.16)
[13]     module.layer2.3.conv1 -- mag_old = 0.0194, mag_now = 0.0209 (1.08)
[14]     module.layer2.3.conv2 -- mag_old = 0.0153, mag_now = 0.0165 (1.08)
[15]     module.layer3.0.conv1 -- mag_old = 0.0180, mag_now = 0.0188 (1.04)
[16]     module.layer3.0.conv2 -- mag_old = 0.0155, mag_now = 0.0163 (1.05)
[16] module.layer3.0.downsample.0 -- mag_old = 0.0270, mag_now = 0.0275 (1.02)
[17]     module.layer3.1.conv1 -- mag_old = 0.0122, mag_now = 0.0132 (1.08)
[18]     module.layer3.1.conv2 -- mag_old = 0.0121, mag_now = 0.0137 (1.13)
[19]     module.layer3.2.conv1 -- mag_old = 0.0124, mag_now = 0.0132 (1.07)
[20]     module.layer3.2.conv2 -- mag_old = 0.0116, mag_now = 0.0130 (1.12)
[21]     module.layer3.3.conv1 -- mag_old = 0.0121, mag_now = 0.0130 (1.07)
[22]     module.layer3.3.conv2 -- mag_old = 0.0107, mag_now = 0.0120 (1.12)
[23]     module.layer3.4.conv1 -- mag_old = 0.0125, mag_now = 0.0132 (1.06)
[24]     module.layer3.4.conv2 -- mag_old = 0.0109, mag_now = 0.0122 (1.12)
[25]     module.layer3.5.conv1 -- mag_old = 0.0129, mag_now = 0.0137 (1.07)
[26]     module.layer3.5.conv2 -- mag_old = 0.0113, mag_now = 0.0122 (1.07)
[27]     module.layer4.0.conv1 -- mag_old = 0.0139, mag_now = 0.0143 (1.03)
[28]     module.layer4.0.conv2 -- mag_old = 0.0108, mag_now = 0.0112 (1.04)
[28] module.layer4.0.downsample.0 -- mag_old = 0.0196, mag_now = 0.0201 (1.03)
[29]     module.layer4.1.conv1 -- mag_old = 0.0102, mag_now = 0.0108 (1.06)
[30]     module.layer4.1.conv2 -- mag_old = 0.0090, mag_now = 0.0096 (1.06)
[31]     module.layer4.2.conv1 -- mag_old = 0.0108, mag_now = 0.0113 (1.05)
[32]     module.layer4.2.conv2 -- mag_old = 0.0084, mag_now = 0.0086 (1.03)
[33]                 module.fc -- mag_old = 0.0460, mag_now = 0.0451 (0.98)
[120331 4071 2022/02/21-15:46:16]    Acc1 65.1520 Acc5 86.7340 Loss_test 1.4198 | Acc1_train -1.0000 Acc5_train -1.0000 Loss_train -1.0000 | Epoch 0 (Best_Acc1 65.1520 @ Best_Acc1_Epoch 0) lr 0.01
[120331 4071 2022/02/21-15:46:16] predicted finish time: only one time stamp, not enough to predict
[120331 4071 2022/02/21-15:46:17] ==> Set lr = 0.01 @ Epoch 1_

After 25 epochs accuracy is still around ~66-67%, and this drop after first epoch doesn't look like something that should happen.

**Comment by MingSun-Tse** [comment]
Hi, @ia3leonidshad , thanks for your interest in our work! (1) "this drop after first epoch doesn't look like something that should happen." -- this is because the regularization in GReg is trained using LR 0.001 (note the `--lr_prune` setting), when switched to LR 0.01 (note `--lr_ft`), it is normal to see accuracy drop. (2) "After 25 epochs accuracy is still around ~66-67%" -- the accuracy should have a major jump after the 1st LR decay (30 epochs), so right now, I can only suggest completing the whole fine-tuning process (with the recommended 90-epoch LR schedule). If you still cannot reproduce the results, let me know!

**Comment by ia3leonidshad** [comment]
I changed `--lr_ft` to start with 0.001 and it resolved the issue, after 30 epochs accuracy restored to baseline model.
